libvast/src/system/partition_transformer.cpp

```diff
@@ -95,91 +99,115 @@ void partition_transformer_state::add_slice(const table_slice& slice) {
 
 void partition_transformer_state::finalize_data() {
   // Serialize the finished indexers.
-  for (auto& [qf, idx] : indexers) {
-    data.indexer_chunks.emplace_back(qf.name(), chunkify(idx));
+  for (auto& [layout, typed_indexers] : indexers)
+    for (auto& [qf, idx] : typed_indexers)
+      data[layout].indexer_chunks.emplace_back(qf.name(), chunkify(idx));
+  for (auto& [layout, partition_data] : data) {
+    auto& mutable_synopsis = partition_data.synopsis.unshared();
+    mutable_synopsis.shrink();
+    // TODO: It would probably make more sense if the partition
+    // synopsis keeps track of offset/events internally.
+    mutable_synopsis.offset = partition_data.offset;
+    mutable_synopsis.events = partition_data.events;
```

Comment:
Can we find another way to group the logic. This can probably be inlined and fused together into the respective loops that come afterwards?